### ISSUE TITLE
Create Page To View All IATI Registry Publisher Accounts With Status of 'Pending'

### ISSUE THREAD

**Issue by wendyrogers** [okfn/ckanext-iati#94]
It would be extremely useful for IATI Registry Admin users to have a page similar to http://iatiregistry.org/dashboard/organizations but that only lists publisher accounts with a status of 'Pending'. Please could you advise how much resource this would take to create?

**Comment by chrisviderum** [comment]
@wendyrogers  would it be fine, if we sort out the other open issues, before we are getting our hands on this one?

**Comment by wendyrogers** [comment]
@chrisviderum yes given that there are other issues more pressing at the moment I am happy for us to look at this one once the backlog of issues relating to the recent Registry upgrade have been cleared

**Comment by pmartignoli** [comment]
Hi @wendyrogers @dalepotter 
Two options are available, both requiring 18 hours for development and testing.  
(1)We can display the publishers with a status of "pending" in the "My Publishers" page (https://iatiregistry.org/dashboard/organizations) and add an "Approved" column, or (2) create an additional page placed as a tab in the dashboard (News Feed, My Datasets, My Publishers, Pending Approval). Please let me know how you would like to proceed.

**Comment by wendyrogers** [comment]
Thank you @pmartignoli . As this function must only be available to 'Admin' users my preference would be for above option 1

**Comment by PetyaKangalova** [comment]
@pmartignoli I would also support option 1 as we are already using the  "My publishers"  page to check for registered accounts, so it will be great to have two separate columns for  "Pending" and "Approved".

**Comment by dalepotter** [comment]
@pmartignoli I agree with @wendyrogers and @PetyaKangalova - let's go ahead with option 1.

Could I suggest we add just one extra column, perhaps titled `Status`.  When completed, the table might look like this:

Publisher	Organisation | Type | HQ Country / Region | Datasets | Status
--- | --- | --- | --- | ---
1%Club | | | 0 | Approved
ACODEV | | | 0 | Approved
Acme Inc | | | 0 | Pending approval

As @wendyrogers mentioned, this extra column should only be visible to admin users.

I noticed that the `Type` and `HQ Country / Region` columns contain no data for any publisher. Is this a bug, or has no publisher entered this infortmation?

**Comment by pmartignoli** [comment]
@dalepotter @PetyaKangalova @wendyrogers 
This is deployed to production. Please test and let me know if there is anything further required. 
https://iatiregistry.org/publisher

**Comment by PetyaKangalova** [comment]
@pmartignoli  Thanks for informing us. I have just checked and noticed two issues:

1. We have only asked for the `Status` column to be shown on this [page](https://iatiregistry.org/dashboard/organizations) so that it is visible for admin users. There should be **no status** on the main public [Registry page](https://www.iatiregistry.org/publisher). Could this be removed as soon as possible? All publishers that are showing on the Registry publisher page are already approved and it will confuse users if there is a status column available there.

2. On the [Registry Dashboard page](https://iatiregistry.org/dashboard/organizations) there are currently no pending accounts. Is this correct? I am sure the are organisations that are pending approval. For instance, this one: https://iatiregistry.org/publisher/edit/wws

**Comment by pmartignoli** [comment]
@PetyaKangalova, changes made last week were reverted. Please test the following format in staging.  https://iati-staging.ckan.io/dashboard/pending

**Comment by PetyaKangalova** [comment]
@pmartignoli Many thanks for updating this in staging. I quite like that you have created a separate tab for "Pending" accounts, even though we asked for a separate column within the main "My Publisher" tab.  I am happy for you to keep it as it is at the moment (having separate tab for "Pending" accounts) as long as we make sure it is updated with any new pending accounts.

On the main "My Publisher" tab, we can now see all approved Registry accounts, which is great. I have two questions:
1. At the moment there is a column for Organisation Type and HQ Country and Region but no information has been pulled through. Is there are a reason why?
2. My second question is more of a request as to whether a column with the **IATI Organisation Identifier (required) *** field can be added to this page. This will be extremely useful as in a lot of cases  need to check that new publishers haven't used the same IATI Organisation identifier.

![image](https://user-images.githubusercontent.com/24607542/27142025-bfe505c2-5121-11e7-80ee-344051803a23.png)

**Comment by pmartignoli** [comment]
@PetyaKangalova The pending datasets tab has been tested and deployed. Each new publisher will be displayed here once created. 

CKAN core was displaying the "approval needed" publishers as "active" in the status column, and  further development would have been required. 

For the two questions.
 
1. The two columns are empty because the helper that is used for retrieving the data is not returning the data. A custom helper and controller would be needed in order to display the Org Type and HQ country. 

2. Yes, it is possible to add an column displaying the required data. But for clarification, will this be required only for the Pending tab?

**Comment by PetyaKangalova** [comment]
@pmartignoli  Many thanks for the clarification. In response to your questions:

1.  I am not sure I understand the process for having or adding

>  a custom helper and controller 
I assumed that if the columns were there already they should be populated with data @dalepotter  would you be able to help?

2. I was referring to the main "My publisher" page. It will be great to have the organisation identifiers on both the "My Publisher" and "Pending" accounts page.
